astropy/io/fits/hdu/base.py

```diff
@@ -380,6 +380,19 @@ def writeto(self, name, output_verify='exception', overwrite=False,
         hdulist.writeto(name, output_verify, overwrite=overwrite,
                         checksum=checksum)
 
+    @classmethod
+    def size_of_header(cls, header):
```

Comment:
Should probably be `data_size` to be consistent with the existing attributes, but there is already a `_data_size` attribute that includes the padding. Should the new classmethod also include the padding ? (on the HDU object this would make sense I guess).

Reply:
Any uses I can think of this would include padding. How many bytes on disk will this file take when written (both data and header components) would be really useful for  me.

Reply:
I will change it to `data_size` and include padding.

Reply:
@aaryapatil - Looking at this PR, I think we should have `Header.data_size` (the one below) and another `data_sized_padded` with the padding. Both are useful. I'm less convinced by the classmethod here so I would remove that. Would you have time to update the PR ?